internal/staticanalysis/parsing/js_parsing.go

```diff
@@ -68,114 +101,124 @@ func runParser(parserPath, jsFilePath, jsSource string, extraArgs ...string) (st
 		}
 	}
 
-	out, err := cmd.Output()
-	if err != nil {
+	if _, err := cmd.Output(); err != nil {
 		return "", err
 	}
 
-	return string(out), nil
+	if output, err := os.ReadFile(outFilePath); err != nil {
+		return "", fmt.Errorf("runParser failed to read output file: %w", err)
+	} else {
+		return string(output), nil
+	}
+}
+
+func (p parseOutputJSON) process() languageResult {
```

Comment:
Can this code has quite a bit of nesting. Can it be simplified by:

1. Having another function that is called per-file; and/or
1. Extracting the code to generate the various fileData types into small methods such as:
  `IdentifierFromToken(Token) identifier`
  `LiteralFromToken(Token) literal`
  etc?

Reply:
Sure, I'll try to break it up to reduce nesting depth

Reply:
I moved the outer loop into the calling function, I think it looks better that way. How does it seem to you now?

Reply:
It does look better. Happy to leave it for this PR.